crates/sdk/src/env/ext.rs

```diff
@@ -1,8 +1,133 @@
-use borsh::to_vec as to_borsh_vec;
+use borsh::{to_vec as to_borsh_vec, to_vec, BorshDeserialize, BorshSerialize};
 
 use super::{expected_boolean, expected_register, panic_str, read_register, DATA_REGISTER};
 use crate::sys;
-use crate::sys::Buffer;
+use crate::sys::{Buffer, BufferMut};
+
+/// A blockchain proposal action.
+///
+/// This enum represents the different actions that can be executed against a
+/// blockchain, and combined into a proposal.
+///
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, Eq, PartialEq)]
+pub enum ProposalAction {
```

Comment:
this should not need to be directly exposed (yet)

```suggestion
enum ProposalAction {
```